### ISSUE TITLE
is work outside app.multithreaded.run() possible?

### ISSUE THREAD

**Issue by LyingDutch** [CrowCpp/crow#145]
Is it possible to do work forever on the background (looping trough a specific vector for example) outside the server, while app.multrihreaded.run() is running? I cant seem to find a solution for this. It seems I can only do work inside the CROW_ROUTE's.

**Comment by The-EDev** [comment]
The only thing that `app.multithreaded(x)` does is distribute incoming connections across `x` number of threads (the threads are created when the app is run). your code that calls `app.multithreaded(x).run()` is still single threaded and will be blocked if you run an infinite loop in it.

If you want to run an infinite loop outside the server in parallel, in my opinion it's best to create a thread specifically for that infinite loop.

Crow doesn't have any built-in way of influencing the app (changing settings or routes) while the application is already running.

If you want it to do something like that dynamically, Feel free to open an issue with your specific suggestion and I'll help out as much as I can.

**Comment by The-EDev** [comment]
Closing this for lack of response. Please reopen if you feel the question was not answered properly.

**Comment by LyingDutch** [comment]
Actually I do have a follow up question. Your suggestion worked as a charm, however, dividing the server and the underlying program in 2 threads brings up a new issue. I cant seem te signal the sever (Ctrl + C) anymore to stop. And I dont want a route to be able to close the server for safety reasons. Is it possible to call the server to stop without a route to do so?

**Comment by The-EDev** [comment]
You'll probably need to look into signal handling, crow does allow you to disable its internal signal handler. but I'm not sure how to do the signal handling itself. I will try to look into how it can be done but it might take time.

**Comment by LyingDutch** [comment]
No hurry, really appreciate the quick responses :)

**Comment by LyingDutch** [comment]
I actually solved the situation, I run it in a docker container. When I run it like this: docker run -p 8000:8000 -it server /bin/bash
I keep control to signal. And I just use a boolean in while loop for the forever work, thats true as long as app.run() is not completed.
